File: CS/WindowsFormsApplication2/XtraForm1.cs
using DevExpress.Pdf;
using DevExpress.XtraBars.Navigation;
using DevExpress.XtraPdfViewer;
using System.Collections.Generic;

namespace WindowsFormsApplication2 {
    public partial class XtraForm1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        List<CommentData> comments = null;
        CommentsPanelHelper helper = null;
        public XtraForm1() {
            InitializeComponent();
            string fileName = "Comments.pdf";
            helper = new CommentsPanelHelper(accordionControl1);
            pdfViewer1.DocumentChanged += PdfViewer1_DocumentChanged;
            pdfViewer1.MouseDown += PdfViewer1_MouseDown;
            accordionControl1.ElementClick += AccordionControl1_ElementClick;
            pdfViewer1.ZoomMode = PdfZoomMode.FitToVisible;

            pdfViewer1.LoadDocument(fileName);
        }

        private void PdfViewer1_DocumentChanged(object sender, PdfDocumentChangedEventArgs e) {
            pdfViewer1.NavigationPaneInitialVisibility = PdfNavigationPaneVisibility.Hidden;
            InitAccordionControl(e.DocumentFilePath);
        }
        private void PdfViewer1_MouseDown(object sender, System.Windows.Forms.MouseEventArgs e) {
            PdfDocumentContent content = pdfViewer1.GetContentInfo(e.Location);
            if (content.ContentType == PdfDocumentContentType.Annotation) {
                CommentData stickyNote = comments.Find((item) => item.BoundingBox.Contains(content.DocumentPosition.Point));
                if (stickyNote == null) return;
                string annotationName = stickyNote.Name;
                AccordionControlElement element = accordionControl1.GetElements().Find((item) => item.Name == annotationName);
                if (element != null)
                {
                    accordionControl1.Tag = element.Name;
                    accordionControl1.Refresh();
                }
            }
        }
        void SelectTextAnnotation(AccordionControlElement element)
        {
            PdfDocumentPosition position = (PdfDocumentPosition)element.Tag;
            pdfViewer1.EnsureVisibility(position);
            pdfViewer1.SelectAnnotation(position.PageNumber, element.Name);
            accordionControl1.Tag = element.Name;
        }
        private void AccordionControl1_ElementClick(object sender, DevExpress.XtraBars.Navigation.ElementClickEventArgs e) {
            AccordionControlElement element = e.Element;
            if (element.Style == ElementStyle.Group)
                SelectTextAnnotation(element);
            else
                SelectTextAnnotation(element.OwnerElement);

        }
        private void InitAccordionControl(string fileName) {
            CommentReader reader = new CommentReader();
            comments = new List<CommentData>();
            comments = reader.ReadComments(fileName);
            helper.ClearPanel();
            helper.InitPanel(comments);
        }
    }
}

File: VB/WindowsFormsApplication2/XtraForm1.vb
Imports DevExpress.Pdf
Imports DevExpress.XtraBars.Navigation
Imports DevExpress.XtraPdfViewer
Imports System.Collections.Generic

Namespace WindowsFormsApplication2
    Partial Public Class XtraForm1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Private comments As List(Of CommentData) = Nothing
        Private helper As CommentsPanelHelper = Nothing
        Public Sub New()
            InitializeComponent()
            Dim fileName As String = "Comments.pdf"
            helper = New CommentsPanelHelper(accordionControl1)
            AddHandler pdfViewer1.DocumentChanged, AddressOf PdfViewer1_DocumentChanged
            AddHandler pdfViewer1.MouseDown, AddressOf PdfViewer1_MouseDown
            AddHandler accordionControl1.ElementClick, AddressOf AccordionControl1_ElementClick
            pdfViewer1.ZoomMode = PdfZoomMode.FitToVisible

            pdfViewer1.LoadDocument(fileName)
        End Sub

        Private Sub PdfViewer1_DocumentChanged(ByVal sender As Object, ByVal e As PdfDocumentChangedEventArgs)
            pdfViewer1.NavigationPaneInitialVisibility = PdfNavigationPaneVisibility.Hidden
            InitAccordionControl(e.DocumentFilePath)
        End Sub
        Private Sub PdfViewer1_MouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs)
            Dim content As PdfDocumentContent = pdfViewer1.GetContentInfo(e.Location)
            If content.ContentType = PdfDocumentContentType.Annotation Then
                Dim stickyNote As CommentData = comments.Find(Function(item) item.BoundingBox.Contains(content.DocumentPosition.Point))
                If stickyNote Is Nothing Then
                    Return
                End If
                Dim annotationName As String = stickyNote.Name
                Dim element As AccordionControlElement = accordionControl1.GetElements().Find(Function(item) item.Name = annotationName)
                If element IsNot Nothing Then
                    accordionControl1.Tag = element.Name
                    accordionControl1.Refresh()
                End If
            End If
        End Sub
        Private Sub SelectTextAnnotation(ByVal element As AccordionControlElement)
            Dim position As PdfDocumentPosition = CType(element.Tag, PdfDocumentPosition)
            pdfViewer1.EnsureVisibility(position)
            pdfViewer1.SelectAnnotation(position.PageNumber, element.Name)
            accordionControl1.Tag = element.Name
        End Sub
        Private Sub AccordionControl1_ElementClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.Navigation.ElementClickEventArgs)
            Dim element As AccordionControlElement = e.Element
            If element.Style = ElementStyle.Group Then
                SelectTextAnnotation(element)
            Else
                SelectTextAnnotation(element.OwnerElement)
            End If

        End Sub
        Private Sub InitAccordionControl(ByVal fileName As String)
            Dim reader As New CommentReader()
            comments = New List(Of CommentData)()
            comments = reader.ReadComments(fileName)
            helper.ClearPanel()
            helper.InitPanel(comments)
        End Sub
    End Class
End Namespace